internal/pkg/provision/providers/docker/docker.go

```diff
@@ -43,9 +44,24 @@ func (p *provisioner) Close() error {
 
 // GenOptions provides a list of additional config generate options.
 func (p *provisioner) GenOptions(networkReq provision.NetworkRequest) []generate.GenOption {
-	return []generate.GenOption{
+	ret := []generate.GenOption{
 		generate.WithPersist(false),
 	}
+
+	if len(networkReq.Nameservers) > 0 {
+		nameservers := make([]string, len(networkReq.Nameservers))
+		for i := range nameservers {
+			nameservers[i] = networkReq.Nameservers[i].String()
+		}
+
+		ret = append(ret, generate.WithNetworkConfig(
```

Comment:
I wonder if we could unify both provisioners now actually?

firecracker passes DNS servers via kernel `ip=` argument, but I think it might just as well use config. DNS shouldn't be needed before config is downloaded.

Reply:
Looked like there were some differences in the `WithNetwork` gen option in firecracker. It also needs the eth0 interface unlike docker.

Reply:
ah, got it, as it's embedded internally...